server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remote/delivery/HeloNameProvider.java

```diff
@@ -41,7 +41,7 @@ public String getHeloName() {
         if (heloName == null) {
             // TODO: Maybe we should better just lookup the hostname via dns
             try {
-                return domainList.getDefaultDomain();
+                return domainList.getDefaultDomain().name();
```

Comment:
asString?

Reply:
In this case the user is supposed to receive the domain in the same case it gave it, so using .name seems a better choice.